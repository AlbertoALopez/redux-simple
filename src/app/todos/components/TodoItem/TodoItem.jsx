/* React component for a single To Do item */
import React from 'react';

const TodoItem = ({
    onClick,
    completed,
    text,
    id
}) => {
    return (
        <li
            onClick={onClick}
            style={{
                textDecoration:
                    completed ?
                    'line-through' :
                    'none',
            }}
        >
            <div className="todo-item-text">{text}</div>
        </li>
    );
};

export default TodoItem;
