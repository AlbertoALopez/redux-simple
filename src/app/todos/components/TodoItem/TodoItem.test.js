import React from 'react';
import renderer from 'react-test-renderer';
import TodoItem from './TodoItem';
import { mount } from 'enzyme';


test('Todo Item renders a list item', () => {
    const component = renderer.create(
        <TodoItem
            text="Test"
            completed={false}
            onClick={() => {
                console.log('test');
            }}
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Todo Item renders text properly', () => {
    const wrapper = mount(
        <TodoItem
            text="Test"
            completed={false}
            onClick={() => {
                console.log('test');
            }}
        />
    );
    const p = wrapper.find('.todo-item-text');
    expect(p.text()).toBe('Test');
});
