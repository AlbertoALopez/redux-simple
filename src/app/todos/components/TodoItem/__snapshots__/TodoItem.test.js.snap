exports[`test Todo Item renders a list item 1`] = `
<li
  onClick={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  }>
  <div
    className="todo-item-text">
    Test
  </div>
</li>
`;
